crates/control/src/role_assignment.rs

```diff
@@ -625,7 +625,9 @@ fn decide_if_claiming_striker_or_other_role(
     filtered_game_controller_state: Option<&FilteredGameControllerState>,
     optional_roles: &[Role],
 ) -> (Role, bool, Option<BallPosition<Field>>) {
-    if time_to_reach_kick_position < spl_message.time_to_reach_kick_position {
+    if time_to_reach_kick_position < spl_message.time_to_reach_kick_position
+        && time_to_reach_kick_position.unwrap_or(Duration::MAX) < Duration::from_secs(1200)
```

Comment:
I think this is not correct. You are comparing options in 682. This is really weird to reason about. This should probably be some `.is_some() && `

Reply:
Again what learned. Fixed